### PR TITLE
Add Helm chart and PR Deployments

### PR DESCRIPTION
**Pull request by kliwniloc**
This Pull Request merges the Helm chart from https://github.com/ls1intum/artemis-helm and adds GitHub action workflows to continuously deploy pull requests to Kubernetes.

## Summary by CodeRabbit

- **New Features**
	- Introduced automated workflows for deploying and cleaning up Kubernetes environments for pull requests.
	- Added a workflow for building and pushing Helm charts on branch pushes.
	- Provided a guide for accessing application URLs based on different service types in the Helm chart.
	- Enhanced Helm chart templating with functions for generating resource names and labels.

- **Chores**
	- Created a `.gitignore` file to exclude unnecessary files from version control.

### CODE REVIEW

File: .github/workflows/helm_pr_deployment.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
+name: Deploy PR to Kubernetes
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
```
Comment (iyannsch) [inline .github/workflows/helm_pr_deployment.yml:5]: According to the GH documentation, those 3 values are consistent with the set of default values. I propose to omit them for clarity reasons here

File: .github/workflows/helm_pr_deployment.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  workflow_run:
+    workflows: ["tests"]
+    branches: [main]
+    types:
+      - completed
```
Comment (iyannsch) [inline .github/workflows/helm_pr_deployment.yml:10]: Does this relate to the workflow defined in `.github/workflows/test.yml`? 
What's the reason for adding a workflow-trigger when your *only* job is conditional to the action having a `pull_request`?

File: .github/workflows/helmchart.yml
Comment (iyannsch) [inline .github/workflows/helmchart.yml:1]: Why do you need this workflow when the exact same steps are taken on `pull_request` anyways?

**Review by iyannsch — CHANGES_REQUESTED** [review]
What a nice idea to put the deployments on K8s! 
I left you a couple of comments in the GitHub action files and briefly looked over the templates too - they are mostly copied from Matthias though, right?

**Comment by krusche** [comment]
Please use the PR description template, follow the naming conventions for PR titles and branch names!

**Pull request closed without merging**